application/common/models/User.php

```diff
@@ -641,6 +641,28 @@ public function getNagState()
         return $this->nagState->getState();
     }
 
+    private static function listUsersWithExternalGroupWith($appPrefix): array
+    {
+        $appPrefixWithHyphen = $appPrefix . '-';
+
+        /** @var User[] $users */
+        $users = User::find()->where(
+            ['like', 'groups_external', $appPrefixWithHyphen]
```

Comment:
Doesn't this need a `%` before and after `$appPrefixWithHyphen`? Otherwise it's only going to find records where `groups_external` is exactly 'groupname-' I think. _...some time later..._ I did some reading in the Yii docs and some other code, and I am still puzzled. It appears that Yii magically adds the %, but it is not exactly documented as such. All that to say is that I think this code is correct, but I don't know why.

Reply:
Yii automatically wraps the `like` comparison's value in `%`s. Here are the docs (though I agree they're not always easy to find):
https://www.yiiframework.com/doc/guide/2.0/en/db-query-builder#operator-format